package.json

```diff
@@ -8,8 +8,9 @@
     "contracts:compile:abi": "typechain --target ethers-v5 --out-dir src/abis/types \"./src/abis/**/*.json\"",
     "contracts:compile:v3": "typechain --target ethers-v5 --out-dir src/types/v3 \"./node_modules/@uniswap/**/artifacts/contracts/**/*[!dbg].json\"",
     "contracts:compile": "yarn contracts:compile:abi && yarn contracts:compile:v3",
-    "relay": "relay-compiler",
-    "graphql:generate": "graphql-codegen --config codegen.yml && yarn relay",
+    "relay": "relay-compiler relay.config.js",
+    "relay-thegraph": "relay-compiler relay_thegraph.config.js",
+    "graphql:generate": "get-graphql-schema https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v3 > ./src/graphql/thegraph/schema/schema.graphql && yarn relay-thegraph && get-graphql-schema $(grep REACT_APP_GQL_API_URL .env.local | cut -d '=' -f2) -h X-API-KEY=$(grep REACT_APP_GQL_API_KEY .env.local | cut -d '=' -f2) > ./src/graphql/data/schema/schema.graphql && yarn relay",
```

Comment:
is this only used on local?

Reply:
@vm I'm not sure what to do abt this part... the data team doesn't want us to expose the api key to the public, so I'm not sure how we're supposed to build outside of local?

Reply:
we have to expose the key

Reply:
imo we should check in the schema so we know if it changes. we can have this script to generate the schema be authenticated

Reply:
@cartcrom this is long enough that I think it warrants its own script (see `prei18n:extract`)

Reply:
would also let you use `dotenv` to grab the values instead of needing to inline all the grep/piping

Reply:
@cmcewen I'm not exactly sure what you mean. Are you saying to keep the schema in the repo (not gitignore it) and then only update it when it changes, rather than pulling it each build?

Reply:
yes i think because it's not a "deterministic" build item because it's pulling from a remote URL, and because it's useful to know when the schema changes, i'd vote to check it into the repo and not gitignore it